### DISCUSSION TITLE
Sending a typed-array-buffer-schema model with an opcode?

### DISCUSSION THREAD

**Discussion by ghost** [geckosio/geckos.io#237] [Help / Questions]
So I'm using geckos.io and typed-array-buffer-schema to send game data (both are amazing libs btw Yandeu!), and I'm trying to figure out how to send more than one model through the raw channel. Best way is to use an opcode to differentiate the different packet types so you can do a Model.fromBuffer to the right Model. I can prepend a byte (opcode) to the buffer. How would you do it?

**Comment by yandeu** [comment]
Thanks :)

I have never sent 2 models at the same time. Yes try to concatinate them with a separator. Or make a combined model?

**Reply by ghost** [reply]
I wonder if the BufferSchema.getIdFromBuffer(buffer) id could be used as an opcode or if there was a way to set the buffer id yourself.

**Reply by yandeu** [reply]
> more than one model through the raw channel

Instead of sending the `Players` model and the `Enemies` model as 2. Just combine them into a `World` model.  

Something like:  
```js
const worldSchema = BufferSchema.schema('world', {
  players: [playerSchema],
  enemies: [enemySchema]
})
```

**Reply by ghost** [reply]
Still trying to figure out how to send more than one schema. Argh! :)
